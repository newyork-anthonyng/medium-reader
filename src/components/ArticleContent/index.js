import React, { PureComponent } from 'react'
import T from 'prop-types'

class ArticleContent extends PureComponent {
  render () {
    if (this.props.isLoading) {
      return <h1>Fetching medium content</h1>
    }

    if (this.props.text) {
      return <p>{this.props.text}</p>
    }

    return null
  }
}

ArticleContent.propTypes = {
  isLoading: T.bool,
  text: T.string
}

export default ArticleContent
