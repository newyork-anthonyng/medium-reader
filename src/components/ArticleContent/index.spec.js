/* global describe, it, expect */

import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import ArticleContent from './'

describe('ArticleContent', () => {
  const defaultProps = {
    isLoading: false,
    text: 'This is my awesome article'
  }

  it('should render correctly when loading', () => {
    const wrapper = shallow(
      <ArticleContent
        {...defaultProps}
        isLoading
      />
    )

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should render correctly when there is text', () => {
    const wrapper = shallow(
      <ArticleContent {...defaultProps} />
    )

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should render correctly when not loading and no text', () => {
    const wrapper = shallow(
      <ArticleContent
        {...defaultProps}
        isLoading={false}
        text=''
      />
    )

    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
