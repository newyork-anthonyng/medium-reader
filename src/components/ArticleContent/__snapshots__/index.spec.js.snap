// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleContent should render correctly when loading 1`] = `
<h1>
  Fetching medium content
</h1>
`;

exports[`ArticleContent should render correctly when not loading and no text 1`] = `null`;

exports[`ArticleContent should render correctly when there is text 1`] = `
<p>
  This is my awesome article
</p>
`;
